static/js/stream_settings_ui.js

```diff
@@ -1009,6 +1009,28 @@ export function hide_or_disable_stream_privacy_options_if_required(container) {
             }
         }
     }
+
+    if (!settings_data.user_can_create_public_streams()) {
+        const public_stream_elem = container.find(
+            `input[value='${CSS.escape(stream_data.stream_privacy_policy_values.public.code)}']`,
+        );
+        public_stream_elem.prop("disabled", true);
+    }
```

Comment:
I guess these don't need to look at whether the value is already checked, since we always offer them?  Makes sense.